### PR TITLE
Feat/obs interpolated position

### PR DESCRIPTION
**Pull request by jerabaul29**
Following recurrent user requests, this PR:

- adds a robust function to interpolate messy / "with holes" time series, e.g. gnss positions; the idea is that real world data are messy and there may be NaNs, NaTs, holes, etc. So these are checked quite carefully, and a "robust best attempt" is done: interpolate as NaN if there are no close data points, as "nearest" if there is close data points only on one side in time, as "linear" if there is close data points on both sides in time
- performs interpolation of the gnss lat and lon to the times of the wave measurements in the omb reader, and include these in the xr produced by the omb reader
- document this in the in-depth omb example

### CODE REVIEW

**Comment by gauteh** [comment]
Hi Jean,

I'm looking at this now and thinking about how to do this so that we get a consistent set of features and methods in TrajAn. It seems that this functionality is very similar to that of `gridtime`, and that `gridtime` perhaps is a special case of interpolating observational data to a time coordinate.

The other question is how much post-processed data to include in these files. The goal should not be a one-liner in trajan in itself. You have to call that one-liner from some python-script anyway, so when you generate data for a particular end-user or purpose it is better to add the specific adaptations there as another line or two and keep the default in trajan as basic as possible. That will make the methods easier to compose into blocks for different purposes. We see in OpenDrift that adding a lot of particular functions or arguments to functions is a never-ending game, the next use case is slightly different and a new function or argument is needed. Better to have the user write a few lines, than one very specific one.

**Comment by knutfrode** [comment]
Hi!
I dit not study in very much detail, but also get the impression that there is probably quite a bit overlap between existing `gridtime` and the new `interpolate_variable_to_newtimes`
And so I share Gautes concern that we might here fall into the trap of Duplication, which is the Root of all Evil:
https://dev.to/ivangavlik/dont-repeat-yourself-practical-advice-at-code-level-1mem
The problem is then that it is very hard to improve on this later, when there will be many internal and external cross-dependencies.

Thus small and well defined methods with very specific and clear task and input-output is desired, rather than larger methods which are very convenient ("oneliner") for a specific task, but just slightly unusable for similar tasks.
The existing gridtime was also made relatively quickly, so it is probably not ideal either.
So maybe we should sit down one day, i.e. a `hackaton` trying to outline a best strategy as clearly as possible?

**Comment by gauteh** [comment]
I don't have the complete overview, but here are my current thoughts so that you don't have to wait for too long for some feedback:

* gridtime and interp are pretty close, they are trying to do something general: interpolate a variable which has an observation dimension rather than a time dimension (so it fits in the seltime, iseltime series)
* gridtime currently tries to interpolate all obs variables on the same dimension as the trajectories, this makes a lot of sense on e.g. temperature.
* for you gridtime doesn't work, the assumptions or methods are insufficient? though I am still a bit hazy on exactly why, but I believe you.

I am wondering if maybe we should rather try to make an `obsinterp` or `obsgrid` method, where gridtime and interp are special cases. This would then be general to any observational data, and you could select a different obs-variable. I think that by default gridtime (or the new obsinterp) should only work on the trajectory variables by default (and then you can specify a list of variables to interpolate in addition, e.g. temperature). In that way it is close to just interpolate non-trajectory, but observational data: ds.imu_waves.traj.obsinterp(...) would have all the context it need to pick out dimension and variable etc. We could have ds.traj.obsinterp(['imu..', 'temperature']) to do multiple variables at once. 

To summarize:
* maybe make an `obsinterp` that is general to whatever obs dimesnion or variable you choose
* obsinterp should be on dataset level and/or on variable level
* don't try to interpolate everything at once (like gridtime does by default)

**Comment by gauteh** [comment]
Ok. I think this should go in a `ds.traj.gridobs(times)` method. The interpolation method should be selected as an argument, yours can be the default. The `times` argument should be the observational time variable, e.g.: ds.imu_times (then I think it contains all the necessary information about dimension etc). 

I can push up a small skeleton, I also added an `to_1d()` method which will be useful when iterating over trajectories.

On another note, the example_indepth_... example is getting pretty long, meaning it is easy to mess it up when changing unrelated parts of the code. Maybe it's better to have shorter example? Makes it easier for the user to find exactly what they are looking for (even if it means some duplication).

File: trajan/utils.py [hunk 1]
```diff
@@ -0,0 +1,98 @@
+import numpy as np
+import scipy as sp
+
+
+def interpolate_variable_to_newtimes(times: np.ndarray[np.datetime64],
+                                     variable_on_times: np.ndarray,
+                                     newtimes: np.ndarray[np.datetime64],
+                                     timegap_max: np.datetime64 = np.timedelta64(3, "h"),
```
Comment (jerabaul29) [inline trajan/utils.py]: @knutfrode

**Comment by jerabaul29** [comment]
(but I may be remembering wrong :) )
